src/lib/parsers/sol-json.c

```diff
@@ -546,3 +546,25 @@ sol_json_double_to_str(const double value, char *buf, size_t len)
 
     return 0;
 }
+
+/* Check if scanner is pointing to a valid string with a json element of
+ * informed type. May or may not be NULL terminated */
+SOL_API bool
+sol_json_validate(struct sol_json_scanner *scanner, enum sol_json_type type)
```

Comment:
I still don't like this function name. It's fine for an internal name, but it's not OK for an exported API: there are way too many things one could validate in JSON.

Be more specific; something like `sol_json_pointing_at_type()`.